envision/server.py

```diff
@@ -263,9 +263,7 @@ def on_close(self):
         del FRAMES[self._simulation_id]
 
     async def on_message(self, message):
-        time_since_epoch = time.time()
-        frame = Frame(timestamp=time_since_epoch, data=message)
-        self._frames.append(frame)
+        self._frames.append(Frame(data=json.loads(message)))
```

Comment:
one interesting note is in the previous implementation, it never had to json loads the message on the server side since it didn't need any info in the message, so it just stored it as `str`. Now we have to json loads the message to get the frame_time which adds some overhead as message can be big.

1 alternative could be to do something like `float(message.split('frame_time')[0].split(xxx))` but I am not sure if that's clean

Reply:
Yeah, good point.  But I don't think the alternative you suggest is desirable (I agree that it's not clean).  I suppose we could use the `ijson` package to just get the timestamp and then break.  (We could force the timestamp to be the first thing in the json by wrapping everything up as an array.)

I want to consider this a bit more...